cmake/FindMagma.cmake

```
@@ -14,25 +14,31 @@ find_library(MAGMA_LIBRARY
   magma
   PATHS
   ${MAGMA_DIR} $ENV{MAGMA_DIR} ${HIOP_MAGMA_DIR}
+  ENV LD_LIBRARY_PATH ENV DYLD_LIBRARY_PATH
   PATH_SUFFIXES
-  lib)
+  lib64 lib)
+
+if(MAGMA_LIBRARY)
+  get_filename_component(MAGMA_LIBRARY_DIR ${MAGMA_LIBRARY} DIRECTORY)
+endif()
 
 find_path(MAGMA_INCLUDE_DIR
   NAMES
   magma.h
   PATHS
-  ${MAGMA_DIR} $ENV{MAGMA_DIR} ${HIOP_MAGMA_DIR}
+  ${MAGMA_DIR} $ENV{MAGMA_DIR} ${HIOP_MAGMA_DIR} ${MAGMA_LIBRARY_DIR}/..
   PATH_SUFFIXES
   include)
 
-set(MAGMA_INCLUDE_DIR ${MAGMA_INCLUDE_DIR} CACHE PATH "Path to magma.h")
-set(MAGMA_LIBRARY ${MAGMA_LIBRARY} CACHE PATH "Path to magma library")
+set(MAGMA_INCLUDE_DIR "${MAGMA_INCLUDE_DIR}" CACHE PATH "Path to magma.h")
+set(MAGMA_LIBRARY "${MAGMA_LIBRARY}" CACHE PATH "Path to magma library")
```

Similar like for METIS and UMFPACK, why not define this at the top of the file and leave variable empty?
```suggestion
set(MAGMA_LIBRARY CACHE PATH "Path to magma library")
```
Then, let `find_libarary` populate it. This avoids using `" "`, which turn paths and pathnames to strings.